libs/sdk-backend-tiger/src/backend/workspace/accessControl/index.ts

```diff
@@ -0,0 +1,105 @@
+// (C) 2022-2023 GoodData Corporation
+import { IWorkspaceAccessControlService } from "@gooddata/sdk-backend-spi";
+import { AssigneeIdentifierTypeEnum, AvailableAssignees } from "@gooddata/api-client-tiger";
+import { TigerAuthenticatedCallGuard } from "../../../types";
+import { ObjRef, AccessGranteeDetail, GranularGrantee, IAvailableAccessGrantee } from "@gooddata/sdk-model";
+import {
+    convertUserAssignee,
+    convertUserGroupAssignee,
+    convertUserPermission,
+    convertUserGroupPermission,
+} from "../../../convertors/fromBackend/AccessControlConverter";
+import { objRefToIdentifier } from "../../../utils/api";
+
+export class TigerWorkspaceAccessControlService implements IWorkspaceAccessControlService {
+    constructor(private readonly authCall: TigerAuthenticatedCallGuard, private readonly workspace: string) {}
+
+    public async getAccessList(sharedObject: ObjRef): Promise<AccessGranteeDetail[]> {
+        const objectId = await objRefToIdentifier(sharedObject, this.authCall);
+        const permissions = await this.authCall((client) => {
+            return client.actions
+                .permissions({ workspaceId: this.workspace, dashboardId: objectId })
+                .then((result) => result.data);
+        });
+
+        return [
+            ...permissions.users.map(convertUserPermission),
+            ...permissions.userGroups.map(convertUserGroupPermission),
+        ];
+    }
+
+    public async grantAccess(sharedObject: ObjRef, grantees: GranularGrantee[]): Promise<void> {
+        return this.changeAccess(sharedObject, grantees);
+    }
+
+    public async revokeAccess(sharedObject: ObjRef, grantees: GranularGrantee[]): Promise<void> {
+        return this.changeAccess(sharedObject, grantees);
```

Comment:
Shouldn't we remove the grantees completely in this case? For removing grantees in the share dialog we would need that, right? Anyway, it looks suspicious if two methods `grantAccess` and `revokeAccess` do the same.

Reply:
Will reset the permissions for `revokeAccess`. Not sure if we should do anything with `grantAccess`. I would rather keep as is.